cupy/statistics/correlation.py

```diff
@@ -48,7 +49,32 @@ def corrcoef(a, y=None, rowvar=True, bias=None, ddof=None):
     return out
 
 
-# TODO(okuta): Implement correlate
+def correlate(a, v, mode='valid'):
+    """Returns the cross-correlation of two 1-dimensional sequences.
+
+    Args:
+        a (cupy.ndarray): first 1-dimensional input.
+        v (cupy.ndarray): second 1-dimensional input.
+        mode (str, optional): `valid`, `same`, `full`
+
+    Returns:
+        cupy.ndarray: Discrete cross-correlation of a and v.
+
+    .. seealso:: :func:`numpy.correlate`
+
+    """
+    if a.ndim != 1 or v.ndim != 1:
+        raise ValueError('object too deep for desired array')
+    method = cupyx.scipy.signal.choose_conv_method(a, v, mode)
+    if method == 'direct':
+        v = v.conj()
+        out = cupy.math.misc._dot_convolve(a, v[::-1], mode)
+    elif method == 'fft':
+        conj_v = cupy.conj(v[::-1])
+        out = cupy.math.misc._fft_convolve(a, conj_v, mode)
```

Comment:
For the simplification.
```suggestion
    if method == 'direct':
        out = cupy.math.misc._dot_convolve(a, v.conj()[::-1], mode)
    elif method == 'fft':
        out = cupy.math.misc._fft_convolve(a, v.conj()[::-1], mode)
```